ui/partials/main/thWatchedRepoNavPanel.html

```diff
@@ -4,6 +4,12 @@
     <span>
       <form role="search" class="form-inline">
 
+        <span class="btn btn-sm btn-view-nav nav-menu-btn" 
+              ng-show="serverChanged"
+              id="revisionChangedLabel"
+              title="New version of Treeherder has been deployed. Reload to pick up changes.">
+          <span class="fa fa-exclamation-circle">&nbsp;Server Revision Changed</span>
+        </span>
```

Comment:
All but 4 addressed. The navbar button is clickable and prompts to reload the page. Changed the text away from "server revision". Closed the span around just the icon. ng-cloak added.

I also added a button into the uib-alert so you can also reload the page from it.